src/Simplify.cpp

```diff
@@ -5828,6 +5829,13 @@ void simplify_test() {
         check(e, e);
     }
 
+    // This expression is used to cause infinite recursion.
+    {
+        Expr e = Broadcast::make(-16, 2) < (ramp(Cast::make(UInt(16), 7), Cast::make(UInt(16), 11), 2) - Broadcast::make(1, 2));
```

Comment:
This problem happens when you have: `a < b - c`, where `a`, `b`, `c` are all constants and `(b - c)` does not simplify. I don't think you can achieve this behavior with a non-vector constant, since `(b - c)` would have simplified to a single constant. Removing the cast inside the ramp and broadcast also makes `(b - c)` simplifiable into `ramp(6, 11, 2)`.